File: 0515-find-largest-value-in-each-tree-row/0515-find-largest-value-in-each-tree-row.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def largestValues(self, root: Optional[TreeNode]) -> List[int]:
        result=[]
        if not root:
            return []
        q=deque([root])
        def maxi(nums):
            re=[r.val for r in nums]
            return max(re)
                    
        while q:
            l=len(q)
            result.append(maxi(q))
            for i in range(l):
                r=q.popleft()
                if r.left:
                    q.append(r.left)
                if r.right:
                    q.append(r.right)
            
        return result

File: 0515-find-largest-value-in-each-tree-row/0515-find-largest-value-in-each-tree-row.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int maxi(Queue<TreeNode> q){
        int maximum=Integer.MIN_VALUE;
        for(TreeNode a:q){
            maximum=Math.max(maximum,a.val);
        }
        return maximum;
    }
    public List<Integer> largestValues(TreeNode root) {
        List<Integer> result=new ArrayList<>();
        if(root==null){
            return result;
        }
        Queue<TreeNode> q=new LinkedList<>();
        q.add(root);
        while(!q.isEmpty()){
            int l=q.size();
            result.add(maxi(q));
            for(int i=0;i<l;i++){
                TreeNode r=q.poll();
                if(r.left!=null){
                    q.add(r.left);
                }
                if(r.right!=null){
                    q.add(r.right);
                }
            }
        }
        return result;
    }
}